src/main.rs

```diff
@@ -15,72 +15,75 @@ fn main() -> io::Result<()> {
-                    } else {
-                        for todo in lines {
-                            println!("{}", todo)
+        match input.trim() {
+            "1" => {
+                match try_read_lines("todo.txt") {
+                    Ok(lines) => {
+                        if lines.is_empty() {
+                            println!("nothing to do for now \n");
+                            io::stdout().flush()?;
+                        } else {
+                            for todo in lines {
+                                println!("{}", todo)
+                            }
                         }
                     }
+                    Err(why) => eprintln!("Error reading file : {}", why),
                 }
-                Err(why) => eprintln!("Error reading file : {}", why),
-            }
-            println!("\n");
-
-            continue;
-        }
-
-        if input.trim() == "2" {
-            let mut new_todo = String::new();
-            println!("please input new todo >> \n");
-            io::stdout().flush()?;
-            io::stdin().read_line(&mut new_todo)?;
+                println!("\n");
 
-            match try_add_lines(&new_todo) {
-                Err(why) => eprintln!("couldn't add {} : {}", new_todo, why),
-                Ok(_) => println!("success!\n"),
+                continue;
             }
-            continue;
-        }
-
-        if input.trim() == "3" {
-            if try_read_lines("todo.txt")?.len() == 0 {
-                println!("Nothing to delete now \n");
+            "2" => {
+                let mut new_todo = String::new();
+                println!("please input new todo >> \n");
                 io::stdout().flush()?;
-            } else {
-                println!("which one? >");
+                io::stdin().read_line(&mut new_todo)?;
 
-                if let Ok(vector) = try_read_lines("todo.txt") {
-                    for todo in vector {
-                        println!("{}", todo)
-                    }
-                } else {
-                    eprintln!("couldn't read file!!!")
+                match try_add_lines(&new_todo) {
+                    Err(why) => eprintln!("couldn't add {} : {}", new_todo, why),
+                    Ok(_) => println!("success!\n"),
                 }
+                continue;
+            }
+            "3" => {
+                if try_read_lines("todo.txt")?.len() == 0 {
+                    println!("Nothing to delete now \n");
+                    io::stdout().flush()?;
+                } else {
+                    println!("which one? >");
 
-                let mut delete_todo = String::new();
+                    if let Ok(vector) = try_read_lines("todo.txt") {
+                        for todo in vector {
+                            println!("{}", todo)
+                        }
+                    } else {
+                        eprintln!("couldn't read file!!!")
```

Comment:
In this case, it shouldn't go through the following process. Am I correct?
```suggestion
                        eprintln!("couldn't read file!!!")
                        continue;
```

Reply:
Yes, you are right, I changed the way to store Vector<String> to variable in order to avoide duplicate error handling